src/log/normal.c

```diff
@@ -105,26 +105,29 @@ void normal_log_post_test(struct criterion_test_stats *stats) {
             = stats->test_status == CR_STATUS_FAILED ? CRITERION_PREFIX_FAIL :
                                              CRITERION_PREFIX_PASS;
 
-    criterion_plog(level, prefix, _(format),
-            stats->test->category,
-            stats->test->name,
-            stats->elapsed_time);
+    if(stats->test_status == CR_STATUS_SKIPPED) {
+        if(!stats->message) {
+            criterion_pinfo(CRITERION_PREFIX_SKIP, _(msg_post_test_skip),
+                    stats->test->category,
+                    stats->test->name);
+        } else {
+            criterion_pinfo(CRITERION_PREFIX_SKIP, _(msg_test_generic),
+                    stats->test->category,
+                    stats->test->name,
+                    stats->message);
+        }
+
+    } else {
+        criterion_plog(level, prefix, _(format),
+                stats->test->category,
+                stats->test->name,
+                stats->elapsed_time);
+    }
 }
 
 void normal_log_post_suite(struct criterion_suite_stats *stats) {
     for (struct criterion_test_stats *ts = stats->tests; ts; ts = ts->next) {
-        if(ts->test_status == CR_STATUS_SKIPPED){
-            if(!ts->message) {
-                criterion_pinfo(CRITERION_PREFIX_SKIP, _(msg_post_test_skip),
-                        ts->test->category,
-                        ts->test->name);
-            } else {
-                criterion_pinfo(CRITERION_PREFIX_SKIP, "%s::%s: %s\n",
-                        ts->test->category,
-                        ts->test->name,
-                        ts->message);
-            }
-
+        if (ts->test_status == CR_STATUS_SKIPPED) {
```

Comment:
I think that this chunk has to go, as it still makes things inconsistent in parallel for skipped tests that have a description.

It would be better to get rid of this entirely and call log(pre_test) in skip_disabled().